api/v1alpha4/types.go

```diff
@@ -507,3 +508,63 @@ type AddressRecord struct {
 	Hostname string
 	IP       string
 }
+
+// CloudProviderConfigOverrides represents the fields that can be overridden in azure cloud provider config.
+type CloudProviderConfigOverrides struct {
+	RateLimits []RateLimitSpec `json:"rateLimits,omitempty"`
```

Comment:
we might also want to add cloudProviderBackoff options here, specifically:


Name | Description | Remark
-- | -- | --
cloudProviderBackoff | Enable exponential backoff to manage resource request retries | Boolean value, default to false
cloudProviderBackoffRetries | Backoff retry limit | Integer value, valid if&nbsp;cloudProviderBackoff&nbsp;is true
cloudProviderBackoffExponent | Backoff exponent | Float value, valid if&nbsp;cloudProviderBackoff&nbsp;is true
cloudProviderBackoffDuration | Backoff duration | Integer value, valid if&nbsp;cloudProviderBackoff&nbsp;is true
cloudProviderBackoffJitter | Backoff jitter | Float value, valid if&nbsp;cloudProviderBackoff&nbsp;is true

Reply:
this can also be done in a separate PR

Reply:
Created [issue](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1382) for tracking.